test/transformers/test_active_space_transformer.py

```diff
@@ -0,0 +1,150 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for the ActiveSpaceTransformer."""
+
+import unittest
+
+from test import QiskitNatureTestCase
+
+import numpy as np
+
+from qiskit_nature import QiskitNatureError
+from qiskit_nature.drivers import PySCFDriver
+from qiskit_nature.transformers import ActiveSpaceTransformer
+
+
+class TestActiveSpaceTransformer(QiskitNatureTestCase):
+    """ActiveSpaceTransformer tests."""
+
+    def test_full_active_space(self):
+        """Test that transformer has no effect when all orbitals are active."""
+        driver = PySCFDriver(atom='H 0 0 0; H 0 0 0.735', basis='sto3g')
+        q_molecule = driver.run()
+
+        trafo = ActiveSpaceTransformer(num_electrons=2, num_orbitals=2)
+        q_molecule_reduced = trafo.transform(q_molecule)
+
+        assert np.allclose(q_molecule_reduced.mo_onee_ints, q_molecule.mo_onee_ints)
+        assert np.allclose(q_molecule_reduced.mo_eri_ints, q_molecule.mo_eri_ints)
+        assert np.isclose(q_molecule_reduced.energy_shift['inactive_energy'], 0.0)
+
+    def test_minimal_active_space(self):
+        """Test a minimal active space manually."""
+        driver = PySCFDriver(atom='H 0 0 0; H 0 0 0.735', basis='6-31g')
```

Comment:
Hopefully the ints don't change over time with any PySCF changes. Normally we had created an HDF5 file of the molecule and used that since it was guaranteed to not change (unless we updated it ourselves or course).

Reply:
I produced some HDF5 files to ensure that we don't run into such a problem. I produced new files in order to be able to test the existence (or lack of) the `mo_occ` attribute of the `QMolecule`.